client/pkg/fileutil/fileutil.go

```diff
@@ -135,3 +136,36 @@ func CheckDirPermission(dir string, perm os.FileMode) error {
 	}
 	return nil
 }
+
+// RemoveTempFile deletes useless temporary file with prefix tmp on an existing dir
+// Returns error if the dir does not exist or remove file fail
+func RemoveTempFile(dir string) error {
+	if !Exist(dir) {
+		return fmt.Errorf("directory %q does not exist", dir)
+	}
+	fileNames, err := ReadDir(dir)
+	if err != nil {
+		return err
+	}
+	lg, _ := zap.NewProduction()
+	if lg == nil {
+		lg = zap.NewExample()
+	}
+	var removeFailedFiles []string
+	for _, fileName := range fileNames {
+		if strings.HasPrefix(fileName, "tmp") {
```

Comment:
`pkg/filtuil` is meant for generic packages, not necessarily for etcd specific operations. I think this code should belong to server/* package?

Reply:
This function  is meant for removing tmp* files in snap dir but also for any dir to delete files prefix with tmp. I felt it could belong to pkg/filtuil or we can place it in etcdserver package server.go or util.go, should you have any better suggestion？

Reply:
Ok, it makes sense to be in `pkg/filutil` since it's using existing functions in that package. But I don't want other people to use this public function and accidentally delete `tmp*` files in their system. Why don't we make this more generic like `matchFunc func(p string) bool`, just like how Go expose file walk package?

Reply:
Done